### PR TITLE
Add uncapturederror test

### PR DESCRIPTION
**Pull request by greggman**
It wasn't clear if I should just put this test in error_scope.spec.ts or put it here where there was a TODO but technically this isn't error scope so ?

Also the constructor is tested in src/webgpu/idl/constructable.spec.ts

This came up because I was working on the webgpu-debug-helper and needed to test some stuff.

<hr>

**Requirements for PR author:**

- [X] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [X] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [X] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [X] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/error_test.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+        },
+      })
+    );
+    assert(device !== null);
+    this._device = device;
+  }
+
+  // Generates an error of the given filter type. For now, the errors are generated by calling a
+  // known code-path to cause the error. This can be updated in the future should there be a more
+  // direct way to inject errors.
```
Comment (kainino0x) [inline src/webgpu/error_test.ts:35]: I know this is old but could you upgrade this to a doc comment?

File: src/webgpu/error_test.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            usage: 0xffff, // Invalid GPUBufferUsage
+          })
+        );
+        break;
+    }
+    // MAINTENANCE_TODO: This is a workaround for Chromium not flushing. Remove when not needed.
+    this.device.queue.submit([]);
+  }
+
+  // Checks whether the error is of the type expected given the filter.
```
Comment (kainino0x) [inline src/webgpu/error_test.ts:66]: same

File: src/webgpu/error_test.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        return error instanceof GPUOutOfMemoryError;
+      case 'validation':
+        return error instanceof GPUValidationError;
+      case 'internal':
+        return error instanceof GPUInternalError;
+    }
+  }
+
+  // Expect an uncapturederror event to occur. Note: this MUST be awaited, because
+  // otherwise it could erroneously pass by capturing an error from later in the test.
```
Comment (kainino0x) [inline src/webgpu/error_test.ts:79]: same

**Review by kainino0x — APPROVED** [review]
LGTM but I think you forgot to upload `src/webgpu/idl/constructable.spec.ts`

**Comment by kainino0x** [comment]
> It wasn't clear if I should just put this test in error_scope.spec.ts or put it here where there was a TODO but technically this isn't error scope so ?

This seems fine, no strong feeling whether it should be in error_scope or not. Kinda ambiguous because it's part of the same system, sort of like the "catch-all" error scope, but since there might end up being other uncapturederror tests in this file someday, separate file makes enough sense.

File: src/webgpu/error_test.ts (hunk 1 above)
Comment (greggman) [reply src/webgpu/error_test.ts:35]: done

File: src/webgpu/error_test.ts (hunk 3 above)
Comment (greggman) [reply src/webgpu/error_test.ts:79]: done

File: src/webgpu/error_test.ts (hunk 2 above)
Comment (greggman) [reply src/webgpu/error_test.ts:66]: done

**Comment by greggman** [comment]
> LGTM but I think you forgot to upload src/webgpu/idl/constructable.spec.ts

`src/webgpu/idl/constructable.spec.ts` already exists and already has a test that creates each error type and a `GPUUncapturedErrorEvent`

**Comment by greggman** [comment]
https://github.com/gpuweb/cts/blob/441db0a23175268d4a49310a9b4d2f0e3c15dc62/src/webgpu/idl/constructable.spec.ts#L48

**Comment by kainino0x** [comment]
Oh, sorry, got it. LGTM then!

**Pull request merged**